src-tauri/src/main.rs

```diff
@@ -20,22 +21,27 @@ use tauri::{
     SystemTrayMenuItem,
 };
 
+#[derive(Serialize)]
+struct FarmerIdentity {
+    public_key: [u8; 32],
+    mnemonic: String,
+}
+
 #[derive(Serialize)]
 struct DiskStats {
     free_bytes: u64,
     total_bytes: u64,
 }
 
 #[tauri::command]
-async fn farming(path: String) -> [u8; 32] {
-    // start farming, and return the public key of the farmer
-    let public_key = farm(path.into(), "ws://127.0.0.1:9944").await;
-    public_key.unwrap()
+async fn farming(path: String) -> FarmerIdentity {
+    let farmer_identity = farm(path.into(), "ws://127.0.0.1:9944").await;
+    farmer_identity.unwrap()
 }
 
 #[tauri::command]
 fn get_disk_stats(dir: String) -> DiskStats {
-    info!("{}", dir);
+    println!("{}", dir);
```

Comment:
Did you actually mean to do this? Looks like a debug message to me at best.

Reply:
Forgot to merge main into this branch... Fixing similar ones as well now